### PR TITLE
app jwt check and payload persist removal

### PR DESCRIPTION
**Pull request by xavluiz**
this PR also has the logic of removing the payload persist and POST logic

### CODE REVIEW

File: src/com/softwareco/intellij/plugin/SoftwareCoUtils.java [hunk 1]
```diff
@@ -724,24 +724,22 @@ public static String getOsUsername() {
...
-        if (serverIsOnline) {
-            long now = Math.round(System.currentTimeMillis() / 1000);
-            String api = "/data/apptoken?token=" + now;
-            SoftwareResponse resp = SoftwareCoUtils.makeApiCall(api, HttpGet.METHOD_NAME, null);
-            if (resp.isOk()) {
-                JsonObject obj = resp.getJsonObj();
-                return obj.get("jwt").getAsString();
-            }
+    public static String getAppJwt() {
+        long now = Math.round(System.currentTimeMillis() / 1000);
```
Comment (mmclead) [inline src/com/softwareco/intellij/plugin/SoftwareCoUtils.java:728]: @bjacobson26 how did we resolve this with vscode?

Comment (xavluiz) [reply src/com/softwareco/intellij/plugin/SoftwareCoUtils.java:728]: vscode uses a jwt decode npm to get the "id" from the jwt. Intellij uses base64Url  to decode the parts.

public static boolean isAppJwt() {
        String jwt = FileManager.getItem("jwt");
        if (StringUtils.isNotBlank(jwt)) {
            String stippedDownJwt = jwt.indexOf("JWT ") != -1 ? jwt.substring("JWT ".length()) : jwt;
            try {
                String[] split_string = stippedDownJwt.split("\\.");
                String base64EncodedBody = split_string[1];

                org.apache.commons.codec.binary.Base64 base64Url = new Base64(true);
                String body = new String(base64Url.decode(base64EncodedBody));
                Map<String, String> jsonMap;

                ObjectMapper mapper = new ObjectMapper();
                // convert JSON string to Map
                jsonMap = mapper.readValue(body,
                        new TypeReference<Map<String, String>>() {
                        });
                Object idVal = jsonMap.getOrDefault("id", null);
                if (idVal != null && Long.valueOf(idVal.toString()).longValue() > 9999999999L) {
                    return true;
                }
            } catch (Exception ex) {}
        }
        return false;
    }

Comment (mmclead) [reply src/com/softwareco/intellij/plugin/SoftwareCoUtils.java:728]: But why are we making the random IDs anymore.  I thought we stopped doing that.

Comment (xavluiz) [reply src/com/softwareco/intellij/plugin/SoftwareCoUtils.java:728]: The plugins use the app jwt to create the user, but we can talk about doing that differently.  This is to ensure the app jwt doesn't live past the short amount of time the user installs the plugin and gets a real JWT as an anonymous user

**Review by brettmstevens7 — APPROVED** [review]

**Pull request merged**